source/common/router/config_impl.cc

```diff
@@ -1148,14 +1149,32 @@ RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const Http::RequestHead
   }
 
   // Check for a route that matches the request.
-  for (const RouteEntryImplBaseConstSharedPtr& route : routes_) {
-    if (!headers.Path() && !route->supportsPathlessHeaders()) {
+  for (auto route = routes_.begin(); route != routes_.end(); ++route) {
```

Comment:
nit: const auto&

Reply:
`route` is an iterator it is incremented on each iteration so it cannot be a const, and `routes_.begin()`returns a temporary so it cannot be bound to a non const reference

Reply:
ah, right, trying to save a copy here, forget it.